Improve PR Dashboard, Integrated GitHubAPI and Export Features, etc.

KevinLlano opened the pull request:
-Made it easier to download pull request data as CSV or JSON.
-Cleaned up the code for better readability.
-Improved the look and feel of the export buttons.
-Fixed small issues to keep everything working smoothly.
-Integrated GitHub API data
-Created JSON and CSV buttons with simple alerts
-Added search functionality for Owner & Repo
-Refactored code somewhat
-Commented out Login, AI ChatBot, and Scroll Down Button to stay focused on MVP only
<img width="1855" height="978" alt="image" src="https://github.com/user-attachments/assets/93efc052-c3b9-4a42-adac-00e73c10a9a3" />

KevinLlano committed:
added GitHubApi.ts file, set up github token

KevinLlano committed:
refactored code, created JSON, CSV buttons

jazxbx commented:
lgmt!

jazxbx reviewed (approved):
lgtm!

jazxbx reviewed (changes requested):
Hello! Looks good overall. I suggest we move the fetching in PRs Page. This page will contain all data,filters, states. This way, dashboard and the other components receive props and only focus on rendering the data.
I was also thinking of making a separate search component instead of placing the search in the section header. We will plug the search component in PRs page. This way, PRs page will handle the search state, owner, repo.

KevinLlano committed:
Refactored PR dashboard architecture, resolve 4 merge conflicts, delivering MVP

KevinLlano commented:
Refactor PR dashboard architecture:
-Centralize all PR state and filter logic in PRsPage
-Made Dashboard presentational, receiving filtered PRs as props
-Update SectionHeader to be lightweight and focused on filters
-Moved action buttons and improve UI/UX layout somewhat
-Fix Footer with correct team member links and flexbox positioning
-Added real-time reload on search input; OPTIONAL reload only on button click
-Resolved 4-5 merge conflicts
-Added comments for clarity and maintainability
